tfjs-backend-cpu/src/kernels/BatchMatMul.ts

```diff
@@ -106,12 +106,19 @@ export function batchMatMul(args: {
               let sum = 0.0;
 
               for (let k = k0; k < kBlock; k++) {
-                const batchOffsetA = Math.min(bi, batchDimA - 1) * aBatch;
-                const batchOffsetB = Math.min(bi, batchDimB - 1) * bBatch;
+                let batchIndexA = bi;
+                let batchIndexB = bi;
+                if (bi >= batchDimA) {
+                  batchIndexA = bi % batchDimA;
+                } else if (bi >= batchDimB) {
+                  batchIndexB = bi % batchDimB;
+                }
```

Comment:
Is the `if` statement here necessary? `batchDim = Math.max(batchDimA, batchDimB)`, and `bi < batchDim`, so it's never the case that `bi >= batchDimA && bi >= batchDimB` (`bi` can only be greater than or equal to one of them (or none of them)), meaning the `else if` case will always run when its condition is true (since the `if` condition will have been false). That lets us rewrite this code as two if statements:
```ts
if (bi >= batchDimA) {
  batchIndexA = bi % batchDimA;
}
if (bi >= batchDimB) {
  batchIndexB = bi % batchDimB;
}
```
Then, we can remove the `if` statements and just use modulus:

```suggestion
                const batchIndexA = bi % batchDimA;
                const batchIndexB = bi % batchDimB;
```

Reply:
Thank you Matt! Done.